suite/volume/volume_setup.go

```diff
@@ -0,0 +1,320 @@
+			}
+			job, err := n.CreateJob(fluxSpec)
+			if err != nil {
+				return err
+			}
+			d.FluxInstancesMu.Lock()
+			d.NodesByHostPort[n.URL()] = n
+			d.ContractToJobsMap[fluxInstance.Address()] = append(d.ContractToJobsMap[fluxInstance.Address()],
+				JobByInstance{
+					ID:       job.Data.ID,
+					Instance: n.URL(),
+				})
+			d.FluxInstancesMu.Unlock()
+		}
+		d.FluxInstancesMu.Lock()
+		*d.FluxInstances = append(*d.FluxInstances, fluxInstance)
+		d.FluxInstancesMu.Unlock()
+		return nil
+	})
+}
+
+// roundsStartTimes gets run start time for every contract
+func (vt *FluxTest) roundsStartTimes() (map[string]float64, error) {
+	mu := &sync.Mutex{}
+	g := &errgroup.Group{}
+	contactsStartTimes := make(map[string]float64)
+	for contractAddr, jobs := range vt.ContractsToJobsMap {
+		contractAddr := contractAddr
+		jobs := jobs
+		g.Go(func() error {
+			startTimesAcrossNodes := make([]int64, 0)
+			for _, j := range jobs {
+				// get node for a job
+				node := vt.NodesByHostPort[j.Instance]
+				runs, err := node.ReadRunsByJob(j.ID)
+				if err != nil {
+					return err
+				}
+				runsStartTimes := make([]int64, 0)
+				for _, r := range runs.Data {
+					runsStartTimes = append(runsStartTimes, r.Attributes.CreatedAt.UnixNano()/1e6) // milliseconds
+				}
+				lastRunStartTime := maxInt64Slice(runsStartTimes)
+				startTimesAcrossNodes = append(startTimesAcrossNodes, lastRunStartTime)
+			}
+			mu.Lock()
+			defer mu.Unlock()
+			// earliest start across nodes for contract
+			contactsStartTimes[contractAddr] = float64(minInt64Slice(startTimesAcrossNodes))
+			return nil
+		})
+	}
+	err := g.Wait()
+	if err != nil {
+		return nil, err
+	}
+	log.Debug().Interface("Round start times", contactsStartTimes).Send()
+	return contactsStartTimes, nil
+}
+
+func (vt *FluxTest) getAllRoundsSubmissionsEvents(ctx context.Context, roundID int, submissions int, submissionVal *big.Int) (map[string]float64, error) {
+	g, gctx := errgroup.WithContext(ctx)
+
+	mu := &sync.Mutex{}
+	endTimes := make(map[string]float64)
+	for _, fi := range *vt.FluxInstances {
+		fi := fi
+		g.Go(func() error {
+			endTime, err := fi.FilterRoundSubmissions(gctx, submissions, submissionVal, roundID)
+			if err != nil {
+				return err
+			}
+			mu.Lock()
+			defer mu.Unlock()
+			endTimes[fi.Address()] = float64(endTime)
+			return nil
+		})
+	}
+	err := g.Wait()
+	if err != nil {
+		return nil, err
+	}
+	return endTimes, nil
+}
+
+// roundsMetrics gets start times from runs via API, awaits submissions for each contract and get round end times
+func (vt *FluxTest) roundsMetrics(roundID int, submissions int, submissionVal *big.Int) error {
+	startTimes, err := vt.roundsStartTimes()
+	if err != nil {
+		return err
+	}
+	endTimes, err := vt.getAllRoundsSubmissionsEvents(context.Background(), roundID, submissions, submissionVal)
+	if err != nil {
+		return err
+	}
+	for contract := range startTimes {
+		duration := endTimes[contract] - startTimes[contract]
```

Comment:
As mentioned this'd be much cleaner if it was subtracting of timestamps resulting in a `time.Duration`.